python/pylibcudf/pylibcudf/tests/test_column_from_array.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import pylibcudf as plc
+
+np = pytest.importorskip("numpy")
+cp = pytest.importorskip("cupy")
+
+NUMPY_DTYPES = [
+    np.int8,
+    np.int16,
+    np.int32,
+    np.int64,
+    np.uint8,
+    np.uint16,
+    np.uint32,
+    np.uint64,
+    np.float32,
+    np.float64,
+    np.bool_,
+]
+
+
+@pytest.fixture(params=NUMPY_DTYPES, ids=repr)
+def np_1darray(request):
+    dtype = request.param
+    return np.array([0, 1, 2, 3], dtype=dtype)
+
+
+@pytest.fixture
+def cp_1darray(np_1darray):
+    return cp.asarray(np_1darray)
+
+
+@pytest.fixture(params=NUMPY_DTYPES, ids=repr)
+def np_2darray(request):
+    dtype = request.param
+    return np.array([[0, 1, 2], [3, 4, 5]], dtype=dtype)
+
+
+@pytest.fixture
+def cp_2darray(np_2darray):
+    return cp.asarray(np_2darray)
+
+
+def test_from_ndarray_cupy_1d(cp_1darray, np_1darray):
+    expect = pa.array(
+        np_1darray, type=pa.from_numpy_dtype(np_1darray.dtype.type)
+    )
+    got = plc.Column.from_array(cp_1darray)
+    assert_column_eq(expect, got)
+
+
+def test_from_ndarray_cupy_2d(cp_2darray, np_2darray):
+    dtype = pa.from_numpy_dtype(np_2darray.dtype.type)
+    expect = pa.array(np_2darray.tolist(), type=pa.list_(dtype))
+    got = plc.Column.from_array(cp_2darray)
+    assert_column_eq(expect, got)
+
+
+def test_from_ndarray_numpy_1d(np_1darray):
+    with pytest.raises(
+        NotImplementedError,
+        match="Converting to a pylibcudf Column is not yet implemented.",
+    ):
+        plc.Column.from_array(np_1darray)
+
+
+def test_from_ndarray_numpy_2d(np_2darray):
+    with pytest.raises(
+        NotImplementedError,
+        match="Converting to a pylibcudf Column is not yet implemented.",
+    ):
+        plc.Column.from_array(np_2darray)
+
+
```

Comment:
Can we add tests validating that we get errors if trying to convert from data that is not c contiguous (if those do not already exist)?

Reply:
done https://github.com/rapidsai/cudf/pull/18311/commits/617efbda436d7529708aee2039ed7c7cd8825599#diff-d81ea7feeafcf39a47f29ed628804e45bcab30c98d83f8df7be36d2757fa3957